### PR TITLE
Rename scanner token methods

### PR DESCRIPTION
**Pull request by sandersn**
Also rename a method from formatting scanner.
This commit doesn't rename internals or usages.

Edit: setTokenEnd actually resets all 3 token positions. I'm still thinking of a better name for it.
Edit: 2: I'm going with resetTokenState

Fixes my inability to remember which scanner method refers to what edge of a token.

### CODE REVIEW

File: src/compiler/parser.ts [hunk 1]
```diff
@@ -2122,11 +2122,11 @@ namespace Parser {
     }
 
     function scanError(message: DiagnosticMessage, length: number): void {
-        parseErrorAtPosition(scanner.getTextPos(), length, message);
+        parseErrorAtPosition(scanner.getTokenEnd(), length, message);
     }
 
     function getNodePos(): number {
-        return scanner.getStartPos();
+        return scanner.getTokenFullStart();
```
Comment (sandersn) [inline src/compiler/parser.ts:2129]: this is unexpected; I would have thought a node would only include its own span, not the whitespace/trivia in front of it.
changing it to getTokenStart breaks every test ever, though -- symbol baselines of course, emit now misses leading comments, and some types are missed somehow.

File: src/compiler/parser.ts [hunk 2]
```diff
@@ -2344,7 +2344,7 @@ namespace Parser {
                 return;
 
             case "is":
-                parseErrorAt(pos, scanner.getTextPos(), Diagnostics.A_type_predicate_is_only_allowed_in_return_type_position_for_functions_and_methods);
+                parseErrorAt(pos, scanner.getTokenEnd(), Diagnostics.A_type_predicate_is_only_allowed_in_return_type_position_for_functions_and_methods);
```
Comment (sandersn) [inline src/compiler/parser.ts]: this is also suspicious
Edit: No difference in tests when changing to getTokenStart, so I'm going with that.

File: src/services/classifier.ts [hunk 3]
```diff
@@ -152,8 +152,8 @@ export function createClassifier(): Classifier {
                 handleToken();
                 lastNonTriviaToken = token;
             }
-            const end = scanner.getTextPos();
-            pushEncodedClassification(scanner.getTokenPos(), end, offset, classFromKind(token), spans);
+            const end = scanner.getTokenEnd();
+            pushEncodedClassification(scanner.getTokenStart(), end, offset, classFromKind(token), spans);
```
Comment (sandersn) [inline src/services/classifier.ts:156]: we have some weird patterns with saving getTokenStart/getTokenEnd into temp variables, even though those functions are stateless.

File: src/services/classifier.ts [hunk 4]
```diff
@@ -989,17 +989,17 @@ export function getEncodedSyntacticClassifications(cancellationToken: Cancellati
...
 
-        while (mergeConflictScanner.getTextPos() < end) {
+        while (mergeConflictScanner.getTokenEnd() < end) {
             classifyDisabledCodeToken();
         }
     }
 
     function classifyDisabledCodeToken() {
-        const start = mergeConflictScanner.getTextPos();
+        const start = mergeConflictScanner.getTokenEnd();
```
Comment (sandersn) [inline src/services/classifier.ts:1000]: sus

(in the same way as the previous code)

**Comment by gabritto** [comment]
@typescript-bot run DT

File: src/services/classifier.ts (hunk 3 above)
Comment (DanielRosenwasser) [reply src/services/classifier.ts:156]: It's inefficient to call scanner methods over and over again needlessly ;)

File: src/compiler/scanner.ts [hunk 5]
```diff
@@ -2623,11 +2635,11 @@ export function createScanner(languageVersion: ScriptTarget,
         languageVariant = variant;
     }
 
-    function setTextPos(textPos: number) {
-        Debug.assert(textPos >= 0);
-        pos = textPos;
-        startPos = textPos;
-        tokenPos = textPos;
+    function resetTokenState(position: number) {
```
Comment (DanielRosenwasser) [inline src/compiler/scanner.ts:2638]: The fact that this hanles positions but talks about token state feels off.

File: src/services/classifier.ts [hunk 6]
```diff
@@ -700,16 +700,16 @@ export function getEncodedSyntacticClassifications(cancellationToken: Cancellati
     }
 
     function classifyLeadingTriviaAndGetTokenStart(token: Node): number {
-        triviaScanner.setTextPos(token.pos);
+        triviaScanner.setTokenEnd(token.pos);
         while (true) {
-            const start = triviaScanner.getTextPos();
+            const start = triviaScanner.getTokenEnd();
```
[the remark this answers is not in the document]
Comment (DanielRosenwasser) [reply src/services/classifier.ts:705]: `getTokenEnd` refers to either `token.pos` as you've initialized above or the end of the last token scanned. So it's suspicious but I believe it's correct.

If you wanted to change things up, I would move this down after the "fast path", scan, and use `getTokenFullStart`.

File: src/compiler/scanner.ts (hunk 5 above)
Comment (sandersn) [reply src/compiler/scanner.ts:2638]: `resetTokenPositions` ? `resetTokenToPosition` ? `resetTokenAtPosition` ? But it also resets token kind, flags and value.

File: src/services/classifier.ts (hunk 3 above)
Comment (jakebailey) [reply src/services/classifier.ts:156]: I hate to suggest it but... what if we destructured these for perf too? 😄

Comment (jakebailey) [reply src/services/classifier.ts:156]: Oh, this is classifier. Probably pointless.

Comment (DanielRosenwasser) [reply src/services/classifier.ts:156]: I was thinking about this recently - it probably makes sense in the parser itself.

**Review by andrewbranch — APPROVED** [review]
This is so good 💯

**Pull request merged**